extensions/amp-viz-vega/0.1/amp-viz-vega.js

```diff
@@ -124,15 +136,13 @@ export class AmpVizVega extends AMP.BaseElement {
    * @private
    */
   renderGraph_() {
-    // TODO(aghassemi): Replace with actual rendering implementation.
     return new Promise((resolve, unused) => {
-      const text = 'To be replaced with Vega graph with data: ' +
-          JSON.stringify(this.data_);
-      vsyncFor(this.win).mutate(() => {
-        const textNode = this.element.ownerDocument.createTextNode(text);
-        this.container_.appendChild(textNode);
+      vega.parse.spec(this.data_, (error, chart) => {
```

Comment:
We are, the `vega.parse` name is misleading. `this.data_` is a parsed JSON object, JSON string comes from either a remote `src` or inline `<script>` and we do the parsing.

That being said, data in Vega spec can also be Urls to other remote Vega spec files and Vega runtime makes the XHR call and does the parsing for those. If that's not something we like, we might be able to manipulate the spec file and inline any Url references by doing the XHR and parsing ourselves.